tests/testsuite/cargo_features.rs

```diff
@@ -113,6 +113,204 @@ release and is no longer necessary to be listed in the manifest
         .run();
 }
 
+#[cargo_test]
+fn allow_features() {
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                cargo-features = ["test-dummy-unstable"]
+
+                [package]
+                name = "a"
+                version = "0.0.1"
+                authors = []
+                im-a-teapot = true
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    // NOTE: We need to use RUSTC_BOOTSTRAP here since we also need nightly rustc
```

Comment:
Instead of using RUSTC_BOOTSTRAP, can these tests be modified to just add a check for `is_nightly` at the top of each test?  I don't think there's really a need to test these on stable.  I feel a little uncomfortable using `RUSTC_BOOTSTRAP` in the testsuite in this way.

Something like: 

```rust
    if !is_nightly() {
        // -Z allow-features is permanently unstable.
        return;
    }
```

Reply:
Is the testsuite automatically run on nightly as well? I wanted to include the test even on stable/beta so that users who work on cargo in that context can still run the tests.

Reply:
On CI, the testsuite runs on stable/beta/nightly.  For any rustc features that are nightly only, the tests are usually disabled on non-nightly toolchains ([example](https://github.com/rust-lang/cargo/blob/3c44c3c4b7900b8b13c85ead25ccaa8abb7d8989/tests/testsuite/rustdoc_extern_html.rs#L47-L50)).  It's not much of an issue if some tests don't run on stable.